File: zhillbot.py
#! /usr/bin/env python3

import sys

from zhill.hill import Hill
from zhill.bot import Bot

if len(sys.argv) != 2:
    print('usage: {0} hill'.format(sys.argv[0]))
    sys.exit(1)

hill = Hill(sys.argv[1])
bot = Bot(hill)
bot.start()

File: zhillbot.rb
#! /usr/bin/env ruby

require_relative 'zhill/hill'
require_relative 'zhill/server'
require_relative 'zhill/bot'

if ARGV.length != 1
  puts "usage: #{$0} hill"
  exit
end

hill_manager = HillManager.new(ARGV[0])
server = Server.new(hill_manager)
bot = Bot.make(hill_manager, server)

bot.start
